### PR TITLE
Gallery page extended

### PR DESCRIPTION
**Pull request by sanjeev-cs**
Changed the gallery page layout and also added the more detail overlay window for a image.

### CODE REVIEW

File: frontend/src/App.jsx
Comment (ConnorBP) [inline frontend/src/App.jsx:1]: Pretty sure You broke the index gallery listing layout here

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -10,11 +10,9 @@ function App() {
   return (
     <BrowserRouter>
       <Routes>
-        <Route path="/" element={<Layout />}>
-          <Route index element={<Gallery />} />
```
Comment (ConnorBP) [inline frontend/src/App.jsx:14]: Wasn't this important? Let's discuss

File: frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+                {/* Right side for the details */}
+                <div className="details_section">
+                    <p><b>Image URL:</b></p>
+                    <input type="text" value={image.imgSrc} readOnly />
+                    <button onClick={copyToClipboard}>{copied ? "Copied!" : "Copy URL"}</button>
+
+                    <p><b>Share:</b></p>
+                    <div className="share_links">
+                        <a href={`https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(image.imgSrc)}`} target="_blank" rel="noopener noreferrer">
```
Comment (ConnorBP) [inline frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx:48]: What's app share links to Facebook. Also double check that these work

File: frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const navigate = useNavigate(); // Navigation hook for programmatic navigation
+    const [copied, setCopied] = useState(false); // State to manage the "Copied" status for the URL
+    const image = images.find((img) => img.id === parseInt(imageId)); // Finding the specific image based on the `imageId`
+
+    if (!image) return <div>Image not found</div>; // If no matching image is found, display a fallback message
+
+    const copyToClipboard = () => {
+        navigator.clipboard.writeText(image.imgSrc);
+        setCopied(true);
+        setTimeout(() => setCopied(false), 2000);
```
Comment (ConnorBP) [inline frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx:20]: What is the purpose of this?

**Review by ConnorBP — CHANGES_REQUESTED** [review]

File: frontend/src/App.jsx (hunk 1 above)
Comment (sanjeev-cs) [reply frontend/src/App.jsx:14]: let me look at it.

File: frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx (hunk 2 above)
Comment (sanjeev-cs) [reply frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx:48]: it's just to share the image on fb

File: frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx (hunk 3 above)
Comment (sanjeev-cs) [reply frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx:20]: The purpose of this was to copy the img url to clipboard.

Comment (ConnorBP) [reply frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx:21]: Yeah but why the timeout

File: frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx (hunk 2 above)
Comment (ConnorBP) [reply frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx]: Look closer at the WhatsApp line. It's also Facebook

**Review by ConnorBP — APPROVED** [review]
Nice, looks good to merge

**Pull request merged**